PWR/PWR_button/Core/Src/main.c

```diff
@@ -0,0 +1,182 @@
+#include "stm32l053xx.h"
+
+#define BUTTON_DEBOUNCE	50
+
+uint8_t flag_EXTI 		= 0,
+		user_bt_count	= 0;	//counter to debounce the button
+
+void MCU_Init(void);
+
+void EXTI4_15_IRQHandler (void)
+{
+	if(EXTI->PR & EXTI_PR_PIF10)	//if there's EXTI interrupt
+	{
+		flag_EXTI = 1;
+		EXTI->PR |= EXTI_PR_PIF10;	//clears the EXTI flag
+	}
+}
+
+int main(void)
+{
+	MCU_Init();
+
+	while(1)
+	{
+		if(flag_EXTI)
+		{
+			if(user_bt_count < 250)					//limits the counting to 250
+			{
+				user_bt_count++;					//counts up for debouncing
+			}
+
+			if((user_bt_count >= BUTTON_DEBOUNCE) && (GPIOA->IDR & GPIO_IDR_ID10_Msk))
+			{
+				GPIOA->ODR &= ~GPIO_ODR_OD7_Msk;	//turn the green led off
+				EXTI->PR |= EXTI_PR_PIF10;			//clears EXTI flag again before entering stop mode
+
+				SCB->SCR = SCB_SCR_SLEEPDEEP_Msk; 	//low-power mode = stop mode
+				PWR->CR &= ~PWR_CR_PDDS;			//making sure we're entering stop mode
+				PWR->CR |= PWR_CR_CWUF;				//clears WUF after 2 system clock cycles
+
+				PWR->CR |= 	PWR_CR_LPSDSR	|		//voltage regulator in low-power mode
+							PWR_CR_ULP		;		//ultra low power mode enable
+				//LPSDSR bit must be set before the LPRUN bit is set
+				PWR->CR |=	PWR_CR_LPRUN;			//voltage regulator in low-power mode
+
+				//entering stop mode procedure
+//				DBGMCU->CR |= DBGMCU_CR_DBG_STOP;	//this bit needs to be set if you're going to debug this code
```

Comment:
As the stop mode retains SRAM and register values, we may also deactivate GPIO's clocks on RCC before entering stop mode. We need to understand a bit further this behaviour, but at a first glance seems to work well and save us some current.

Reply:
I'm not really sure about this. The first time I've tried it I couldn't wake the uC up, and now it is working. But, I had to enable the RCC for GPIOA right after waking up, differently of what you did.

Reply:
Yeah, I forgot to put the line that enables GPIOA clock when wake up on the print that I sent you. 
So enabling the GPIO clock after waking up works?

Reply:
@rodrigocelmer any updates on this?

Reply:
I'm sorry, I forgot to look at this. It looks like it's working without any problems, I just have to enable the clock for GPIOA after waking up

Reply:
Nice